Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1481,6 +1481,40 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+
+    if let modifiers = node.modifiers, modifiers.hasError {
+      for modifier in modifiers {
+        if let detail = modifier.detail, detail.detail != .keyword(.set) {
+          let unexpectedBetweenLeftParenAndDetail = detail.unexpectedBetweenLeftParenAndDetail?.tokens(viewMode: .all)
+          let unexpectedBetweenDetailAndRightParen = detail.unexpectedBetweenDetailAndRightParen?.tokens(viewMode: .all)
+          let unexpectedTokens = (unexpectedBetweenLeftParenAndDetail?.map { $0 } ?? []) + (unexpectedBetweenDetailAndRightParen?.map { $0 } ?? [])
```

Comment:
Didn't know if there was a more nice way to get all unexpected tokens.

Reply:
I think conceptually this is the right approach because you only want the unexpected tokens between the parentheses. I think you can slightly simplify it code-wise to

```swift
[detail.unexpectedBetweenLeftParenAndDetail, detail.unexpectedBetweenDetailAndRightParen]
  .compactMap { $0?.tokens(viewMode: .all) }
```